lib/packages/core/src/service/burstService.ts

```diff
@@ -111,4 +113,52 @@ export class BurstService {
         }
         return response;
     }
+
+    /**
+     * Automatically selects the best host, according to its response time, i.e. the fastest node host will be returned (and set as nodeHost internally)
+     * @param reconfigure An optional flag to set automatic reconfiguration. Default is `true`
+     * Attention: Reconfiguration works only, if you use the default http client. Otherwise, you need to reconfigure manually!
+     * @param checkMethod The optional API method to be called. This applies only for GET methods. Default is `getBlockchainStatus`
+     * @throws Error If `trustedNodeHosts` is empty
+     */
+    public async selectBestHost(reconfigure = false, checkMethod = 'getBlockchainStatus'): Promise<string> {
+        if (!this.settings.trustedNodeHosts.length) {
+            throw new Error('No trustedNodeHosts configured');
+        }
+        const checkEndpoint = this.toBRSEndpoint(checkMethod);
+        let timeout = null;
+        const requests = this.settings.trustedNodeHosts.map(host => {
+            const absoluteUrl = host.endsWith('/') ? `${host}${checkEndpoint}` : `${host}/${checkEndpoint}`;
+            return new Promise<string>(async (resolve, reject) => {
+                try {
+                    await this.settings.httpClient.get(absoluteUrl);
+                    resolve(host);
+                } catch (e) {
+                    if (timeout) {
+                        // @ts-ignore
+                        clearTimeout(timeout);
+                    }
+                    // @ts-ignore
+                    timeout = setTimeout(() => {
```

Comment:
Deferring failed request... assuming that server responses longer than 10 seconds are not desirable at all